components/contribution-flow/index.js

```diff
@@ -968,12 +978,14 @@ class CreateOrderPage extends React.Component {
   }
 
   render() {
-    const { loadingLoggedInUser, LoggedInUser } = this.props;
-
+    const { contributeAs, loadingLoggedInUser, LoggedInUser } = this.props;
+    const steps = this.getSteps();
+    const currentStepName =
+      contributeAs && contributeAs === get(this.state, 'stepProfile.slug') ? steps[1].name : this.props.step;
```

Comment:
This is not the best approach because the route won't reflect the step. We have a different URL for each step `(:step(contributeAs|details|payment|summary))`. You can use the `pushStepRoute` to change the step.

Reply:
Perfect!